src/openforms/submissions/models/submission.py

```diff
@@ -347,6 +347,24 @@ def __str__(self):
             started=localize(localtime(self.created_on)) or _("(no timestamp yet)"),
         )
 
+    def save(self, *args, **kwargs):
+        update_fields = kwargs.get("update_fields")
+        save_all_fields = update_fields is None
+        _will_save_backend_key = (
+            save_all_fields or "finalised_registration_backend_key" in update_fields
+        )
+        if (
+            _will_save_backend_key
+            and self.finalised_registration_backend_key
+            and not self.completed_on
```

Comment:
How is it possible that `self.finalised_registration_backend_key` is set (with `self.completed_on` falsy) in the first place? This seems to contradict with the `ValueError` message below, no?

Reply:
Ah no makes sense, `self.finalised_regisration_backend_key` was set before saving

Reply:
indeed, it's set from the logic action evaluation. tracking that one down was not fun